controller/src/beerocks/master/db/network_map.cpp

```diff
@@ -497,10 +497,20 @@ void network_map::send_bml_bss_tm_req_message_to_listeners(db &database,
         return;
     }
 
-    auto event            = (BML_EVENT *)response->buffer(0);
-    event->type           = BML_EVENT_TYPE_BSS_TM_REQ;
-    auto size             = sizeof(BML_EVENT);
-    event->data           = GET_MESSAGE_POINTER(BML_EVENT_BSS_TM_REQ, response->buffer(0), size);
+    auto event = (BML_EVENT *)response->buffer(0);
+    if (event == nullptr) {
```

Comment:
I don't know how relevant this check is, because there is no way that event can be nullptr. It can only happen if init() wasn't called (or the argument to buffer() is larger than the allocated size of buffer, but since it's 0 that can't happen).

But OK, no harm I guess.

Reply:
And here as well - please change to `if (!event)`

Reply:
But it is not a smart pointer, so it is not relevant.
Now that I look at it, I notice that the cast need to be changed to a reinterpret_cast.

Reply:
should have been a (nitpick)